shell/platform/darwin/ios/framework/Source/FlutterViewControllerTest.mm

```diff
@@ -190,6 +195,145 @@ - (void)testStartKeyboardAnimationWillInvokeSetupKeyboardAnimationVsyncClient {
+
+  // Start show keyboard animation.
+  CGRect initialShowKeyboardBeginFrame = CGRectMake(0, screenHeight, 0, 0);
+  CGRect initialShowKeyboardEndFrame = CGRectMake(0, screenHeight - 250, 0, 0);
+  NSNotification* fakeNotification = [NSNotification
+      notificationWithName:UIKeyboardWillHideNotification
+                    object:nil
+                  userInfo:@{
+                    @"UIKeyboardFrameBeginUserInfoKey" : @(initialShowKeyboardBeginFrame),
+                    @"UIKeyboardFrameEndUserInfoKey" : @(initialShowKeyboardEndFrame),
+                    @"UIKeyboardAnimationDurationUserInfoKey" : @(0.25),
+                    @"UIKeyboardIsLocalUserInfoKey" : @(isLocal)
+                  }];
+  viewControllerMock.targetViewInsetBottom = 100;
+  [viewControllerMock handleKeyboardNotification:fakeNotification];
+  BOOL isShowingAnimation1 = [viewControllerMock keyboardAnimationIsShowing];
+  XCTAssertTrue(isShowingAnimation1 == YES);
+
+  // Start compounding show keyboard animation.
+  CGRect compoundingShowKeyboardBeginFrame = CGRectMake(0, screenHeight - 250, 0, 0);
+  CGRect compoundingShowKeyboardEndFrame = CGRectMake(0, screenHeight - 500, 0, 0);
+  fakeNotification = [NSNotification
+      notificationWithName:UIKeyboardWillHideNotification
+                    object:nil
+                  userInfo:@{
+                    @"UIKeyboardFrameBeginUserInfoKey" : @(compoundingShowKeyboardBeginFrame),
+                    @"UIKeyboardFrameEndUserInfoKey" : @(compoundingShowKeyboardEndFrame),
+                    @"UIKeyboardAnimationDurationUserInfoKey" : @(0.25),
+                    @"UIKeyboardIsLocalUserInfoKey" : @(isLocal)
+                  }];
+
+  viewControllerMock.targetViewInsetBottom = 200;
+  [viewControllerMock handleKeyboardNotification:fakeNotification];
+  BOOL isShowingAnimation2 = [viewControllerMock keyboardAnimationIsShowing];
+  XCTAssertTrue(isShowingAnimation2 == YES);
+  XCTAssertTrue(isShowingAnimation1 == isShowingAnimation2);
+
+  // Start hide keyboard animation.
+  CGRect initialHideKeyboardBeginFrame = CGRectMake(0, screenHeight - 500, 0, 0);
+  CGRect initialHideKeyboardEndFrame = CGRectMake(0, screenHeight - 250, 0, 0);
+  fakeNotification = [NSNotification
+      notificationWithName:UIKeyboardWillHideNotification
+                    object:nil
+                  userInfo:@{
+                    @"UIKeyboardFrameBeginUserInfoKey" : @(initialHideKeyboardBeginFrame),
+                    @"UIKeyboardFrameEndUserInfoKey" : @(initialHideKeyboardEndFrame),
+                    @"UIKeyboardAnimationDurationUserInfoKey" : @(0.25),
+                    @"UIKeyboardIsLocalUserInfoKey" : @(isLocal)
+                  }];
+
+  viewControllerMock.targetViewInsetBottom = 100;
+  [viewControllerMock handleKeyboardNotification:fakeNotification];
+  BOOL isShowingAnimation3 = [viewControllerMock keyboardAnimationIsShowing];
+  XCTAssertTrue(isShowingAnimation3 == NO);
```

Comment:
Nits:
1. If you want to assert a false value, better to use `XCTAssertFalse` directly. ^_^
2. For `BOOL` I think it's better to use `XCTAssertTrue(isShowingAnimation2);` instead of `XCTAssertTrue(isShowingAnimation2 == YES);`

Reply:
Thought so too, but copied from another test case. Changed back to this way